File: contributed-source-code/shootout/strcat/strcat.lua
-- $Id$
-- http://shootout.alioth.debian.org
-- contributed by Roberto Ierusalimschy

local n = tonumber((arg and arg[1]) or 1)
local buff = {}
for i=1,n do
  table.insert(buff, "hello\n")
end
local s = table.concat(buff)
print(string.len(s))

File: contributed-source-code/shootout/strcat/strcat.php
#!/usr/bin/php -f
<?php
/*
 $Id$
 http://shootout.alioth.debian.org/
*/
$n = ($argc == 2) ? $argv[1] : 1;
$str = "";
while ($n-- > 0) {
    $str .= "hello\n";
}
$len = strlen($str);
print "$len\n";
?>
